sdks/corda/src/main/kotlin/com/weaver/corda/sdk/InteroperableHelper.kt

```diff
@@ -214,7 +215,8 @@ class InteroperableHelper {
             externalStateLinearId: String
         ): String {
             val responseView = getExternalStateView(proxy, externalStateLinearId)
-            return responseView.payload.toStringUtf8()
+            val interopPayload = InteropPayloadOuterClass.InteropPayload.parseFrom(responseView.notarizedPayloadsList[0].payload)
```

Comment:
The data that it is reading from vault, is already verified as part of proof verification in writeExternalState, (which is written as ExternalState in vault), this helper function is to just read the verified (ExternalState) payload from vault. 
So it may not be necessary to match again, and trust that whatever is written to vault is consistent?

What we can do is change the structure of ExternalState that is written to vault, so that it only contains one payload to avoid confusion, but not sure if its worth to do that

Reply:
OK, let's leave it as is.